From 1e73bdfc923a6f211832dab70c5ee2842d1828a8 Mon Sep 17 00:00:00 2001
From: Alwin Esch <alwin.esch@web.de>
Date: Thu, 12 Mar 2020 00:07:58 +0100
Subject: [PATCH] [vfs][addons] fix title show of compressed package dir

On requests 17066 (Matrix) and 17497 (Leia) was this changed to use the
whole item from addon, but seen now that it remove then before defined
list item title.

This Partially revert the previous change and set now only path like before.
The rest stay onchanged and work confirmed.

This much nowhere else since many places took the path when title was empty.
However, there are also places that absolutely need the title.
---
 xbmc/filesystem/FileDirectoryFactory.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xbmc/filesystem/FileDirectoryFactory.cpp b/xbmc/filesystem/FileDirectoryFactory.cpp
index 9b9d700b418e..1078f1aa2d9c 100644
--- a/xbmc/filesystem/FileDirectoryFactory.cpp
+++ b/xbmc/filesystem/FileDirectoryFactory.cpp
@@ -95,7 +95,7 @@ IFileDirectory* CFileDirectoryFactory::Create(const CURL& url, CFileItem* pItem,
             else
             {
               // compressed or more than one file -> create a dir
-              *pItem = wrap->m_items;
+              pItem->SetPath(wrap->m_items.GetPath());
             }
 
             // Check for folder, if yes return also wrap.
